src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -996,16 +996,17 @@ else if (conf.commitlog_segment_size.toKibibytes() < 2 * conf.max_mutation_size.
         {
             conf.jmx_server_options = JMXServerOptions.createParsingSystemProperties();
         }
-        else if (JMXServerOptions.isEnabledBySystemProperties())
+        else
         {
-            throw new ConfigurationException("Configure either jmx_server_options in cassandra.yaml and comment out " +
-                                             "configure_jmx function call in cassandra-env.sh or keep cassandra-env.sh " +
-                                             "to call configure_jmx function but you have to keep jmx_server_options " +
-                                             "in cassandra.yaml commented out.");
+            if (JMXServerOptions.isEnabledBySystemProperties())
+                throw new ConfigurationException("Configure either jmx_server_options in cassandra.yaml and comment out " +
+                                                 "configure_jmx function call in cassandra-env.sh or keep cassandra-env.sh " +
+                                                 "to call configure_jmx function but you have to keep jmx_server_options " +
+                                                 "in cassandra.yaml commented out.");
+            else
+                conf.jmx_server_options.jmx_encryption_options.applyConfig();
```

Comment:
@smiklosovic if we move this `applyConfig()` call inside the `else` the way it is now, it misses to apply the config for the `createParsingSystemProperties()` if path. Isn't it?

Reply:
no, because if you look closely, it it will eventually call "apply" in build()

Reply:
Oh okay got it. Inside the parsing System properties it is calling the build() which I didn't notice.